lib/order.rb

```diff
@@ -0,0 +1,79 @@
+require_relative "takeaway"
+require_relative "message"
+
+class Order
+  attr_reader :basket, :menu, :total
+
+  def initialize
+    @basket = Hash.new(0)
+    @total = 0
+    @menu = Takeaway.new.menu_list
+  end
+
+  def make_order
+    loop do
+      puts "Here is the menu: #{@menu}"
+      puts "What would you like to order? "
+      selection = gets.chomp
+        if selection.empty?
+          order_summary
+          confirm
+          break
+        elsif selection == "quit"
+          break
+        else 
+          puts "How many would you like to order? "
+          quantity = gets.chomp.to_i
+          add_item(selection, quantity)
+        end  
+    end
+  end
+
+  def add_item(item, quantity)
```

Comment:
good work, this took me so long to figure out. You could set your quantity to a default of 1 so if user doesn't enter how many you have a default value. 

 def add_item(item, quantity = 1)